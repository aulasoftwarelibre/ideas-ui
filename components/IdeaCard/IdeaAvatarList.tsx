import React from "react";
import { User } from "../../model/user/auth";

export interface Props {
  attendess: User[];
}

export const IdeaAvatarList: React.FunctionComponent<Props> = ({
  attendess,
}) => {
  const avatars = attendess
    .slice(0, 4)
    .map((attendee, index) => (
      <img
        className="-ml-1 first:ml-0 inline-block h-8 w-8 rounded-full text-white shadow-solid"
        src={attendee.image ?? "/avatar.svg"} 
        alt=""
        key={index}
      />
    ));
  return (
    <div className="flex overflow-hidden p-1">
      {avatars}
      <div className="font-bold text-white text-base ml-3 mt-1">
        {attendess.length}
      </div>
    </div>
  );
};

export default IdeaAvatarList;
