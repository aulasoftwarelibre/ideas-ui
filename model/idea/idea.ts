import { User } from "../user/auth";

export interface Idea {
  title: string;
  group: string;
  category: string;
  image?: string;
  startsAt: Date;
  online: boolean;
  attendess: User[];
}
