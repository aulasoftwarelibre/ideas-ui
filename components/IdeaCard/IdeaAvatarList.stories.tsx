import React from "react";

import { withKnobs, object } from "@storybook/addon-knobs";

import { User } from "../../model/user/auth";
import IdeaAvatarList from "./IdeaAvatarList";
import faker from "faker";

export default {
  title: "Ideacard/IdeaAvatarList",
  component: IdeaAvatarList,
  decorators: [withKnobs],
};

const avatarAttendees: User[] = [
  { image: faker.image.avatar() },
  { image: faker.image.avatar() },
  { image: faker.image.avatar() },
  { image: faker.image.avatar() },
];
const emptyAvatarAttendees: User[] = [{}, {}, {}, {}];

export const AvatarAttendees = () => (
  <IdeaAvatarList attendees={avatarAttendees}></IdeaAvatarList>
);

export const NoAvatarAttendees = () => (
  <IdeaAvatarList attendees={emptyAvatarAttendees}></IdeaAvatarList>
);

export const Empty = () => <IdeaAvatarList attendees={[]}></IdeaAvatarList>;
