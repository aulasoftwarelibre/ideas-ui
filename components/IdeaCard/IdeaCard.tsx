import classnames from "classnames";
import React from "react";
import Moment from "react-moment";

import { Idea } from "../../model/idea/idea";
import IdeaAvatarList from "./IdeaAvatarList";

export interface Props {
  idea: Idea;
  onClick?: () => void;
}

export const IdeaCard: React.FunctionComponent<Props> = ({ idea, onClick }) => {
  const ideaImage = idea.image ?? "/ideas-banner.png";

  return (
    <div
      className="h-auto max-w-sm rounded-large overflow-hidden shadow-lg "
      onClick={onClick}
    >
      <div
        className="flex h-56 bg-cover bg-center"
        style={{
          backgroundImage: `linear-gradient(to bottom,rgba(0, 0, 0, 0),rgba(0, 0, 0, 1)), url(${ideaImage})`,
        }}
      >
        <div className="p-4 flex-col items-end text-white mt-auto">
          <h4 className="font-bold text-lightgray text-base">
            {idea.category}
          </h4>
          <h3 className="font-bold text-white text-2xl">{idea.title}</h3>
        </div>
        <div className="flex-col items-end text-white mb-auto ml-auto">
          <div
            className={classnames(
              "h-8 px-4 py-1 bg-gradient-l-primary-to-light rounded-tr-large rounded-bl-large",
              { "w-26": idea.online },
              { "w-30": !idea.online }
            )}
          >
            <p className="font-bold text-white text-l">
              {idea.online ? "ONLINE" : "PRESENCIAL"}
            </p>
          </div>
        </div>
      </div>
      <div className="px-5 py-4 bg-gradient-l-primary-to-light">
        <div className="font-bold text-white text-xl capitalize-date">
          <Moment>{idea.startsAt}</Moment>
        </div>
        <div className="font-bold text-white text-sm mb-2">{idea.group}</div>
        <IdeaAvatarList attendees={idea.attendees}></IdeaAvatarList>
      </div>
    </div>
  );
};

export default IdeaCard;
