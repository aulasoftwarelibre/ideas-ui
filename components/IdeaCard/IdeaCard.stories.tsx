import React from "react";

import { withKnobs, object } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";

import IdeaCard from ".";
import { Idea } from "../../model/idea/idea";

export default {
  title: "Ideacard/IdeaCard",
  component: IdeaCard,
  decorators: [withKnobs],
};

const todayDate = Date.now();

const idea: Idea = {
  title: "Idea title",
  group: "Organization",
  category: "Category",
  startsAt: new Date(todayDate),
  online: true,
  attendees: [{}, {}, {}, {}, {}],
  image:
    "https://images.unsplash.com/photo-1486312338219-ce68d2c6f44d?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&ixid=eyJhcHBfaWQiOjF9",
};

export const Online = () => (
  <IdeaCard idea={object("Idea", idea)} onClick={action("clicked")}></IdeaCard>
);

export const Presencial = () => (
  <IdeaCard
    idea={object("Idea", { ...idea, online: false })}
    onClick={action("clicked")}
  ></IdeaCard>
);
