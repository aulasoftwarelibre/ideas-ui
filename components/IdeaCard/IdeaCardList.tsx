import React from 'react';

import { Idea } from '../../model/idea/idea';
import IdeaCard from './IdeaCard';

export interface Props {
  ideas: Idea[];
}

const NotFound: React.FunctionComponent = () => (
  <div className="flex flex-col w-1/2 mx-auto" data-testid="not-found">
    <div className="mx-auto">
      <svg
        className="h-24 w-24"
        xmlns="http://www.w3.org/2000/svg"
        fill="none"
        viewBox="0 0 24 24"
        stroke="currentColor"
      >
        <path
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          d="M8 7V3m8 4V3m-9 8h10M5 21h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v12a2 2 0 002 2z"
        />
      </svg>
    </div>
    <div className="text-center">
      <h2 className="font-bold">
        No se han encontrado coincidencias con tu búsqueda
      </h2>
      <p className="font-light">Prueba otra búsqueda o intentalo más tarde.</p>
    </div>
  </div>
);

export const IdeaCardList: React.FunctionComponent<Props> = ({ ideas }) => {
  if (ideas.length === 0) {
    return <NotFound />;
  }

  return (
    <div className="flex flex-wrap">
      {ideas.map((idea: Idea) => (
        <div key={idea.id} className="w-full md:w-1/2 lg:w-1/3 xl:w-1/4 p-2">
          <IdeaCard idea={idea} />
        </div>
      ))}
    </div>
  );
};

export default IdeaCardList;
