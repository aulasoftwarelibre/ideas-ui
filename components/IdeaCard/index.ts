import IdeaCard from './IdeaCard';

export default IdeaCard;
