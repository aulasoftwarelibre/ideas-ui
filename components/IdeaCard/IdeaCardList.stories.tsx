import React from 'react';

import IdeaCardList, { Props } from './IdeaCardList';
import { ideas } from './IdeaCardList.test';

export default {
  component: IdeaCardList,
  title: "IdeaCard/IdeaCardList",
  argTypes: {
    width: {
      control: {
        type: "range",
        min: 0,
        max: 8,
        step: 1,
      },
    },
  },
};

const Template = (args: Props) => <IdeaCardList {...args} />;

export const Default = Template.bind({});
Default.args = {
  ideas: ideas,
};

export const Empty = Template.bind({});
Empty.args = {
  ideas: [],
};
