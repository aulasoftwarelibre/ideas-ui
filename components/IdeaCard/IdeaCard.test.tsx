import { text } from "@storybook/addon-knobs";
import { shallow } from "enzyme";
import React from "react";
import { Idea } from "../../model/idea/idea";
import IdeaCard from "./IdeaCard";

const todayDate = Date.now();

describe("ActivityCard", () => {
  const onClickMock = jest.fn();

  const idea: Idea = {
    title: "Activity title",
    group: "Organization",
    category: "Category",
    startsAt: new Date(todayDate),
    online: true,
    attendess: [],
    image: "https://source.unsplash.com/featured/?tech",
  };
  const component = shallow(
    <IdeaCard idea={idea} onClick={onClickMock}>
      Activity Card
    </IdeaCard>
  );

  it("should be rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("should be clickable", () => {
    component.simulate("click");

    expect(onClickMock).toHaveBeenCalled();
  });
});
