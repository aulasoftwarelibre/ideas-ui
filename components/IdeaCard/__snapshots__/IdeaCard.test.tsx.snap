// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaCard should be rendered correctly 1`] = `ShallowWrapper {}`;
